### PR TITLE
make aws conditional

### PR DESCRIPTION
**Pull request by simongottschlag**
This PR fixes the issue where gitops-v2 requires AWS azure extension even though AWS isn't used.

### CODE REVIEW

File: gitops-v2/new/push_aws.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    inputs:
+      workingDirectory: "."
+      awsCredentials: ${{ parameters.awsServiceConnection }}
+      regionName: ${{ parameters.awsRegion }}
+      scriptType: inline
+      inlineScript: |
+        set -e
+        SHA_SHORT=$(shaShort)
+
+        aws ecr describe-repositories --repository-names ${SERVICE_NAME} || aws ecr create-repository --repository-name ${SERVICE_NAME} --image-scanning-configuration scanOnPush=true
```
Comment (bittrance) [inline gitops-v2/new/push_aws.yaml:21]: Is it really a good idea to auto-create repos? It seems a big exposure for a small gain. In my opinion, the repo should be part of the application's IaC. I'm even skeptical to granting `describe` perms to Azure as that allows the craxor to discover new attack surfaces.

File: gitops-v2/new/push_aws.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        ECR_URI=$(aws ecr describe-repositories --repository-names ${SERVICE_NAME} | jq -r '.repositories[0].repositoryUri')
+        ECR_PASSWORD=$(aws ecr get-login-password --region ${AWS_REGION})
+        docker login -u AWS -p $ECR_PASSWORD $ECR_URI
```
Comment (bittrance) [inline gitops-v2/new/push_aws.yaml:24]: I think you can `aws ecr get-login-password --region ${AWS_REGION} | docker login -u AWS --password-stdin` so as not to expose the password.

**Review by bittrance — CHANGES_REQUESTED** [review]
Please don't use env vars for secrets.

File: gitops-v2/new/push_aws.yaml (hunk 1 above)
Comment (simongottschlag) [reply gitops-v2/new/push_aws.yaml:22]: This should be handled in separate PR, only moving AWS to make it possible to run without the extension.

File: gitops-v2/new/push_aws.yaml (hunk 2 above)
Comment (simongottschlag) [reply gitops-v2/new/push_aws.yaml:25]: This should be handled in separate PR, only moving AWS to make it possible to run without the extension.

Comment (simongottschlag) [reply gitops-v2/new/push_aws.yaml]: Fixed.

**Review by bittrance — APPROVED** [review]
:+1:

**Pull request merged**